practice-1/main.cpp

```diff
@@ -45,28 +37,26 @@ class Song : public Music {
     const string name;
     unsigned int duration;
 
-    Song() {
-        genre = "";
-        name = "";
-        duration = 0;
-    };
+    Song() : Music(), name(""), genre(""), duration(0) {};
 
-    Song(string _genre, string _name, unsigned int _duration) {
-        genre = _genre;
-        name = _name;
-        duration = _duration;
-    };
+    Song(string _music_id,
+         string _artist_name,
+         unsigned int _year,
+         string _name,
+         string _genre,
+         unsigned int _duration) :
+            Music(_music_id, _artist_name, _year),
+            name(std::move(_name)),
+            genre(std::move(_genre)),
+            duration(_duration) {};
 
     bool operator==(Song &s) {
-        Music music = static_cast<Music>(this);
```

Comment:
This is actually a bug. Since you're casting "this", in lines 66–68 you're comparing the Song (cast as a music object) to itself (as a song object).

Actually, you don't even need to do any casting because the Song already has access to the proper fields to compare.